artifactory/repositories.go

```diff
@@ -95,44 +95,59 @@ func (l LocalRepository) String() string {
+	FailedRetrievalCachePeriodSecs    *int                    `json:"failedRetrievalCachePeriodSecs,omitempty"`
+	MissedRetrievalCachePeriodSecs    *int                    `json:"missedRetrievalCachePeriodSecs,omitempty"`
+	UnusedArtifactsCleanupEnabled     *bool                   `json:"unusedArtifactsCleanupEnabled,omitempty"`
+	UnusedArtifactsCleanupPeriodHours *int                    `json:"unusedArtifactsCleanupPeriodHours,omitempty"`
+	AssumedOfflinePeriodSecs          *int                    `json:"assumedOfflinePeriodSecs,omitempty"`
+	FetchJarsEagerly                  *bool                   `json:"fetchJarsEagerly,omitempty"`
+	FetchSourcesEagerly               *bool                   `json:"fetchSourcesEagerly,omitempty"`
+	ShareConfiguration                *bool                   `json:"shareConfiguration,omitempty"`
+	SynchronizeProperties             *bool                   `json:"synchronizeProperties,omitempty"`
+	BlockMismatchingMimeTypes         *bool                   `json:"blockMismatchingMimeTypes,omitempty"`
+	AllowAnyHostAuth                  *bool                   `json:"allowAnyHostAuth,omitempty"`
+	EnableCookieManagement            *bool                   `json:"enableCookieManagement,omitempty"`
+	BowerRegistryURL                  *string                 `json:"bowerRegistryUrl,omitempty"`
+	ComposerRegistryURL               *string                 `json:"composerRegistryUrl,omitempty"`
+	PyPIRegistryURL                   *string                 `json:"pyPIRegistryUrl,omitempty"`
+	VcsType                           *string                 `json:"vcsType,omitempty"`
+	VcsGitProvider                    *string                 `json:"vcsGitProvider,omitempty"`
+	VcsGitDownloadUrl                 *string                 `json:"VcsGitDownloadUrl,omitempty"`
+	BypassHeadRequests                *bool                   `json:"bypassHeadRequests,omitempty"`
+	ClientTLSCertificate              *string                 `json:"clientTlsCertificate,omitempty"`
+	ExternalDependenciesEnabled       *bool                   `json:"externalDependenciesEnabled,omitempty"`
+	ExternalDependenciesPatterns      *[]string               `json:"externalDependenciesPatterns,omitempty"`
+	DownloadRedirect                  *bool                   `json:"downloadRedirect,omitempty"`
+	FeedContextPath                   *string                 `json:"feedContextPath,omitempty"`
+	DownloadContextPath               *string                 `json:"downloadContextPath,omitempty"`
+	V3FeedUrl                         *string                 `json:"v3FeedUrl,omitempty"`
+	XrayIndex                         *bool                   `json:"xrayIndex,omitempty"`
+	ContentSynchronisation            *contentSynchronisation `json:"contentSynchronisation,omitempty"`
+}
+
+// ContentSynchronisation represents smart remote repository configuration
+type contentSynchronisation struct {
```

Comment:
struct should be exported

```suggestion
type ContentSynchronisation struct {
```

Reply:
Happy to change. This was intentional as I didn't think it should be accessed outside of the package, i.e. it should only accessible via the ContentSynchronisation field of the remoteRepository struct.

Reply:
> This was intentional as I didn't think it should be accessed outside of the package

I do agree that the field shouldn't really be accessed outside the package.

**However**, we currently have established a pattern where we still make it exported. Here are some samples:

* https://github.com/target/go-arty/blob/master/artifactory/artifacts.go#L42-L44
* https://github.com/target/go-arty/blob/master/artifactory/build.go#L31-L42
* https://github.com/target/go-arty/blob/master/artifactory/licenses.go#L52-L54

The simplest reason I can give for why we have this pattern is mainly due to the **potential** for a consumer of this library to want to directly access that field vs using the getter (i.e. `GetContentSynchronisation()`).

Based off that, I think it'd be great if you could make the struct exported 👍

Reply:
No problem at all. :)